### PR TITLE
webdriver: unhandled promise rejection

### LINKED ISSUE #3539: Unhandled promise rejection
**Environment (please complete the following information):**
 - **WebdriverIO version:** v5.4.17
 - **Mode:** standalone
 - **If WDIO Testrunner, running sync/async:** n/a
 - **Node.js version:** 8.9.4
 - **NPM version:** 6.4.1
 - **Browser name and version:** n/a
 - **Platform name and version:** n/a
 - **Additional wdio packages used (if applicable):** n/a

**Describe the bug**
When a custom command throws an exception it produces `UnhandledPromiseRejectionWarning` to the console:

```
(node:8292) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: for test
(node:8292) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**To Reproduce**

```js
const { remote } = require('webdriverio');

(async () => {
  const browser = await remote({
    logLevel: 'trace',
    capabilities: {
      browserName: 'chrome'
    }
  });

  browser.addCommand('shouldThrow', async () => {
    throw new Error('for test');
  });

  try {
    await browser.shouldThrow();
  } finally {
    await browser.deleteSession();
  }
})().catch(e => console.error(e));

```

**Expected behavior**
This code should not produce any warnings

**Additional context**
According to my research, [this promise](https://github.com/webdriverio/webdriverio/blob/master/packages/webdriver/src/monad.js#L105) stays unhandled when is rejected. Adding a catch block there fixed the warning for my code example.

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Fix #3539 

suppress unhandled promise rejection.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/webdriver/src/monad.js [hunk 1]
```diff
@@ -105,6 +105,8 @@ export default function WebDriver (options, modifier, propertiesObject = {}) {
             Promise.resolve(result).then((res) => {
                 log.info('RESULT', res)
                 this.emit('result', { name, result: res })
+            }).catch(err => {
```
Comment (mgrybyk) [inline packages/webdriver/src/monad.js]: adding catch here is not affecting error handling at all, it doesn't matter if sync is on or off.

**Comment by mgrybyk** [comment]
@christian-bromann this catch is only for unhandled promise rejection, all the errors are thrown as expected :)

**Comment by christian-bromann** [comment]
> this catch is only for unhandled promise rejection

I still wonder if we can run in any edge cases with this which would make it super hard to find this out. Anyway no objection to the change if it fixes the problem described.

**Comment by mgrybyk** [comment]
If anyone is able to get to that catch let me know. I was not even able to write test for that catch :(

**Pull request merged**